processing/src/main/java/io/druid/query/extraction/ExplicitDimRenameFn.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.base.Throwables;
+import io.druid.jackson.DefaultObjectMapper;
+
+import javax.annotation.Nullable;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/**
+ *
+ */
+public class ExplicitDimRenameFn extends DimExtractionFn
+{
+  @JsonIgnore
+  private final ObjectMapper mapper;
+
+  private final Map<String, String> renames;
+  private final Function<String, String> extractionFunction;
+  private final boolean retainMissingValue;
+  private final String replaceMissingValueWith;
+  private final ExtractionType extractionType;
+
+  @JsonCreator
+  public ExplicitDimRenameFn(
+      @JsonProperty("renames")
+      final Map<String, String> renames,
+      @Nullable @JsonProperty(value = "retainMissingValue", required = false)
```

Comment:
jackson doesn't support `required` anyways (at least it didn't last time I looked), they really are there more as a coding hint, but if they are distracting I can remove them.